### PR TITLE
[ROCm] Update to use ROCm 4.0 (when building TF with --config=rocm)

### PR DESCRIPTION
**Pull request by deven-amd**
/cc @cheshire @chsigg @nvining-work

### CODE REVIEW

**Review by cheshire — APPROVED** [review]

**Comment by deven-amd** [comment]
@gbaned anything I can do on my end to get this PR merged?

**Comment by gbaned** [comment]
> @gbaned anything I can do on my end to get this PR merged?

@deven-amd This PR is waiting for internal approval. Nothing to do at this moment from your end. Thank you!

**Pull request merged**
